libs/vkd3d/command.c

```diff
@@ -6416,16 +6419,20 @@ static void STDMETHODCALLTYPE d3d12_command_list_CopyResource(d3d12_command_list
 
 static void d3d12_command_list_end_batch(struct d3d12_command_list *list)
 {
+    struct d3d12_command_list_barrier_batch barriers;
     switch (list->transfer_batch.batch_type)
     {
         case VKD3D_BATCH_TYPE_COPY_BUFFER_TO_IMAGE:
         case VKD3D_BATCH_TYPE_COPY_IMAGE_TO_BUFFER:
         case VKD3D_BATCH_TYPE_COPY_IMAGE:
             d3d12_command_list_end_current_render_pass(list, false);
+            d3d12_command_list_barrier_batch_init(&barriers);
             for (size_t i = 0; i < list->transfer_batch.batch_len; i++)
-                d3d12_command_list_before_copy_texture_region(list, &list->transfer_batch.batch[i]);
+                d3d12_command_list_before_copy_texture_region(list, &barriers, &list->transfer_batch.batch[i]);
+            d3d12_command_list_barrier_batch_end(list, &barriers);
```

Comment:
Don't we need an init after the end here?

Reply:
I don't think so, since batch_end basically resets it to the fresh state after flush.